exonum-java-binding-common/src/main/java/com/exonum/binding/common/message/BinaryTransactionMessage.java

```diff
@@ -49,13 +49,15 @@
 
     this.rawTransaction = ByteBuffer.allocate(messageSize).order(LITTLE_ENDIAN);
     this.rawTransaction.put(slice);
+    resetPosition();
   }
 
   @Override
   public PublicKey getAuthor() {
```

Comment:
Do methods that modifies BB state need to be thread safe?

Reply:
That's a good question. As this class feels like a value, one might expect it to be. We can achieve that by duplicating the buffer before doing any operations that require modifying the buffer marks?